homeassistant/components/adguard/__init__.py

```diff
@@ -123,12 +123,7 @@ async def refresh(call: ServiceCall) -> None:
 async def async_unload_entry(hass: HomeAssistant, entry: AdGuardConfigEntry) -> bool:
     """Unload AdGuard Home config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-    loaded_entries = [
-        entry
-        for entry in hass.config_entries.async_entries(DOMAIN)
-        if entry.state == ConfigEntryState.LOADED
-    ]
-    if len(loaded_entries) == 1:
+    if not hass.config_entries.async_loaded_entries(DOMAIN):
```

Comment:
Should these not be adjusted to keep the services registered?

Reply:
It's a common pattern to deregister services when there's no longer any loaded config entry, do you mean it's not a wanted pattern?
If we want to change that, it's for a different set of PRs.

Reply:
I think it's an old pattern we are moving away from

Reply:
OK, then I suggest we make that clear in the developer documentation (if that's not already the case) and remove the unnecessary cleanup in a different set of PRs.

Reply:
Martin pointed to this rule in the quality scale: https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/action-setup/